only prepend `http://` to endpoint if no protocol was provided

Crunch09 opened the pull request:
Hey,
this addresses the fix mentioned in #174: Although the Toxiproxy HTTP API does not support HTTPS, it should not prepend `http://` to the endpoint.

```go
client.NewClient("https://example.com")
// Before: http://https//example.com
// After: https://example.com
```
I would love to write a test for this but `endpoint` is not an exported identifier so i'm not sure how to test this as this is basically the first Go code that i have ever written 🙂 

fixes #174

jpittis commented on client/client.go:
```diff
@@ -47,7 +47,8 @@ type Proxy struct {
 // not overriden)
 func NewClient(endpoint string) *Client {
-	if !strings.HasPrefix(endpoint, "http://") {
+	protocolProvided, _ := regexp.MatchString(`^https?://`, endpoint)
```
Generally it's better to create a regex via `regexp.MustCompile()` and then calling the `MatchString` method. This ensures we don't ignore a possible error in the regex.

Crunch09 commented:
@jpittis Thank you for the review! 👍  I updated the code to now fail when `https://` is provided and removed the regexp again because i felt it didn't add much value then but thank you for letting me know about `MustCompile`! 💯  

**Before**:
![bildschirmfoto 2017-10-12 um 23 17 07](https://user-images.githubusercontent.com/570608/31522053-dec98d38-afa3-11e7-8569-e128ba1beefc.png)

**After**:
![bildschirmfoto 2017-10-12 um 23 17 14](https://user-images.githubusercontent.com/570608/31522060-e535bb92-afa3-11e7-9723-cee49567403b.png)

jpittis commented:
Thanks for the contribution! :)
